### PR TITLE
More Stacks, StackPlans support

### PR DESCRIPTION
**Pull request by brandonc**
I was unable to add meaningful tests yet but plan to backfill them later once the GAP on staging is able to execute stack operations once again.

### CODE REVIEW

File: stack.go [hunk 1]
```diff
@@ -61,6 +64,14 @@ type StackList struct {
 	Items []*Stack
 }
 
+type StackDiagnosticsList struct {
+	*Pagination
+	Items []*StackDiagnostics
+}
+
+type StackDiagnostics struct {
+}
```
Comment (sebasslash) [inline stack.go:73]: Intentionally empty struct?

Comment (brandonc) [reply stack.go:73]: oops!

File: stack.go [hunk 2]
```diff
@@ -82,7 +93,56 @@ type Stack struct {
...
+
+type StackConfigurationStatusTimestamps struct {
+	QueuedAt     *time.Time `jsonapi:"attr,queued-at,omitempty,rfc3339"`
+	CompletedAt  *time.Time `jsonapi:"attr,completed-at,omitempty,rfc3339"`
+	PreparingAt  *time.Time `jsonapi:"attr,preparing-at,omitempty,rfc3339"`
+	EnqueueingAt *time.Time `jsonapi:"attr,enqueueing-at,omitempty,rfc3339"`
+	CanceledAt   *time.Time `jsonapi:"attr,canceled-at,omitempty,rfc3339"`
+	ErroredAt    *time.Time `jsonapi:"attr,errored-at,omitempty,rfc3339"`
+}
```
Comment (sebasslash) [inline stack.go:108]: Why the more constrained format `rfc3339` and not `iso8601`?

Comment (brandonc) [reply stack.go:108]: For all practical purposes, the API emits dates that fits both formats. For the purposes of parsing dates from HCP Terraform, they should be equivalent. RFC3339 is the slightly more stringent version so I usually choose it

File: stack.go [hunk 3]
```diff
@@ -110,6 +161,21 @@ type StackUpdateOptions struct {
 	VCSRepo     *StackVCSRepo `jsonapi:"attr,vcs-repo,omitempty"`
 }
 
+func (s stacks) UpdateConfiguration(ctx context.Context, stackID string) (*Stack, error) {
```
Comment (sebasslash) [inline stack.go:164]: Nit: while we'll never mutate `stacks`, we should define these methods as pointer receivers for consistency.

File: stack_configuration.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	Read(ctx context.Context, id string) (*StackConfiguration, error)
+}
+
+type stackConfigurations struct {
+	client *Client
+}
+
+var _ StackConfigurations = &stackConfigurations{}
+
+func (s stackConfigurations) Read(ctx context.Context, id string) (*StackConfiguration, error) {
```
Comment (sebasslash) [inline stack_configuration.go:24]: We should add a GODOC for all these individual methods

**Review by sebasslash — DISMISSED** [review]
Some minor nits

File: stack_integration_test.go [hunk 5]
```diff
@@ -166,10 +167,156 @@ func TestStackReadUpdateDelete(t *testing.T) {
...
+	for finished := false; !finished; {
+		t.Log("...")
+		select {
+		case <-ctx.Done():
+			t.Fatalf("Stack deployment %s/%s did not have status %q at deadline", stackID, deploymentName, status)
+		case <-ticker.C:
+			var err error
+			deployment, err := client.StackDeployments.Read(ctx, stackID, deploymentName)
+			if err != nil {
+				t.Fatalf("Failed to read stack %q: %s", stackID, err)
```
Comment (Uk1288) [inline stack_integration_test.go:237]: wdyt?

```suggestion
				t.Fatalf("Failed to read stack deployment %s/%s: %s", stackID,  deploymentName, err)
```

File: stack_integration_test.go [hunk 6]
```diff
@@ -166,10 +167,156 @@ func TestStackReadUpdateDelete(t *testing.T) {
...
+			if stackConfig.Status == status {
+				finished = true
+			}
+		}
+	}
+
+	return
+}
+
+func TestStackConverged(t *testing.T) {
```
Comment (Uk1288) [inline stack_integration_test.go:281]: 🎉

**Review by Uk1288 — DISMISSED** [review]

**Review by sebasslash — APPROVED** [review]

**Pull request merged**